contracts/contracts/strategies/FrxEthRedeemStrategy.sol

```diff
@@ -0,0 +1,250 @@
+        }
+
+        require(
+            frxETH.balanceOf(address(this)) == 0,
+            "Not all FraxEth sent to redemption queue"
+        );
+        outstandingRedeems += frxETHStart; // Single set for gas reasons
+
+        // This strategy claims to support WETH, so it is posible for
+        // the vault to transfer WETH to it. This returns any deposited WETH
+        // to the vault so that it is not lost for balance tracking purposes.
+        uint256 wethBalance = weth.balanceOf(address(this));
+        if (wethBalance > 0) {
+            // slither-disable-next-line unchecked-transfer
+            weth.transfer(vaultAddress, wethBalance);
+        }
+
+        emit Deposit(address(frxETH), address(redemptionQueue), frxETHStart);
+    }
+
+    /**
+     * @notice Withdraw an asset from the underlying platform
+     * @param _recipient Address to receive withdrawn assets
+     * @param _asset Address of the asset to withdraw
+     * @param _amount Amount of assets to withdraw
+     */
+    function withdraw(
+        // solhint-disable-next-line no-unused-vars
+        address _recipient,
+        // solhint-disable-next-line no-unused-vars
+        address _asset,
+        // solhint-disable-next-line no-unused-vars
+        uint256 _amount
+    ) external override onlyVault nonReentrant {
+        // Does nothing - all redeems need to be called manually by the
+        // strategist via redeemTickets
+        require(false, "use redeemTickets() instead");
+    }
+
+    /**
+     * @notice Redeem specific tickets from the Queue.
+     * Called by the strategist.
+     * @param _nftIds Array of NFT IDs to redeem
+     */
+    function redeemTickets(uint256[] memory _nftIds, uint256 expectedAmount)
+        external
+        nonReentrant
+    {
+        require(
+            msg.sender == IVault(vaultAddress).strategistAddr(),
+            "Caller is not the Strategist"
+        );
+        uint256 startingBalance = payable(address(this)).balance;
+        for (uint256 i = 0; i < _nftIds.length; i++) {
+            uint256 nftId = _nftIds[i];
+            redemptionQueue.burnRedemptionTicketNft(
+                nftId,
+                payable(address(this))
+            );
+            emit RedeemNFTBurned(nftId);
+        }
+
+        uint256 currentBalance = payable(address(this)).balance;
+        uint256 redeemedAmount = currentBalance - startingBalance;
+        require(
+            expectedAmount == redeemedAmount,
+            "Redeemed amount does not match expected amount"
+        );
+        outstandingRedeems -= redeemedAmount;
+        weth.deposit{ value: currentBalance }();
+        // slither-disable-next-line unchecked-transfer
+        weth.transfer(vaultAddress, currentBalance);
+        emit Withdrawal(
+            address(weth),
+            address(redemptionQueue),
+            currentBalance
+        );
+    }
+
+    function _abstractSetPToken(address _asset, address _pToken)
```

Comment:
I've added a revert on _abstractSetPToken which will prevent a pToken set from happening. I've not adjusted remove, since if there are no adds a remove should not work.